src/Workspaces/CSharp/Portable/Utilities/CSharpUseImplicitTypeHelper.cs

```diff
@@ -259,9 +259,31 @@ protected override bool ShouldAnalyzeForEachStatement(ForEachStatementSyntax for
             }
 
             // variables declared using var cannot be used further in the same initialization expression.
-            if (initializer.DescendantNodesAndSelf()
-                    .Where(n => (n as IdentifierNameSyntax)?.Identifier.ValueText.Equals(identifier.ValueText) == true)
-                    .Any(n => semanticModel.GetSymbolInfo(n, cancellationToken).Symbol?.IsKind(SymbolKind.Local) == true))
+            bool isReusedInInitializer(SyntaxNode n)
+            {
+                // naming collision prerequisite
+                if (!(n is IdentifierNameSyntax id
+                    && id.Identifier.ValueText.Equals(identifier.ValueText)))
+                {
+                    return false;
+                }
+
+                // case of variable direct use: int x = x * 2;
+                if (semanticModel.GetSymbolInfo(n, cancellationToken).Symbol?.IsKind(SymbolKind.Local) == true)
+                {
+                    return true;
+                }
+                // case of qualification starting with the variable name: SomeEnum SomeEnum = SomeEnum.EnumVal1;
+                // note that: SomeEnum SomeEnum = global::SomeEnum.EnumVal1; // is ok and 'var' can be offered
+                // https://github.com/dotnet/roslyn/issues/26894
+                if (n.Parent is MemberAccessExpressionSyntax memberAccessParent && memberAccessParent.Expression == n)
+                {
+                    return true;
+                }
+
+                return false;
```

Comment:
i'm finding all the 'return trues' interspersed with 'return falses' to be confusing.  IMO, this wuld be clearer if one of these forms only occurred once at the end.

Reply:
Works for me 😄